actors/market/src/lib.rs

```diff
@@ -531,92 +531,182 @@ impl Actor {
     }
 
     /// Activate a set of deals, returning the combined deal space and extra info for verified deals.
-    fn activate_deals(
+    /// Returns a vector of activation results parallel and corresponding to the params. Sectors
+    /// empty of deals are implicitly considered activated and return an empty ActivateDealsResult,
```

Comment:
> In order to more easily preserve a parallel mapping between params and results, callers (miner actor) of BatchActivateDeals no longer check for sectors with zero deals. 

I should have read this more carefully.  I think I just have a design disagreement here.  IMO Separation of concerns is more important than ease of caller code though I can see the argument against since market + miner actors are very coupled.

Reply:
I do think that ease of caller code is an ok choice here. Complexity here could lead to hard-to-see bugs. The market actor should be able to handle empty sectors, the only concern I potentially see is an optimisation of sending a few extra bytes. I expect the case of some sectors with deals and some without would not be very common, so optimising it isn't worth much.

I think you and I share a vision that this coupling should eventually go away, and in the resulting code it will be easier to avoid this redundant transmission.

Reply:
> I think you and I share a vision that this coupling should eventually go away, and in the resulting code it will be easier to avoid this redundant transmission.

No argument there

> The market actor should be able to handle empty sectors, the only concern I potentially see is an optimisation of sending a few extra bytes

What I was more worried about is exposing the idea of sector to things like the market which I've been thinking should ideally only care and therefore know about deals.  But maybe knowing about sectors is an ok pattern and even belongs in future market-like actors.

Reply:
> But maybe knowing about sectors is an ok pattern and even belongs in future market-like actors.

Yes, at the moment I think it must. The designs for direct FIL+ even store the sector ID in market state. The sector<->deal mapping must go somewhere, and if the miner is to be ignorant then a market needs to keep it.  (c.f. the verified registry claim too).